config/v1/types_infrastructure.go

```diff
@@ -500,6 +507,37 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// DNSConfigurationType contains information about API and API-Int Load Balancers
+// and who is responsible for that configuration.
+// +union
+type DNSConfigurationType struct {
```

Comment:
To enable external DNS, I think we are only concerned about using coreDNS to run ingress. Should that be taken into account for the shape of this API? Any concerns there?

Reply:
I think this comment was unclear. My concern is that this shape may give the impression that this is the DNS config for everything (API & Ingress) but I believe our current direction is that this will only treat ingress.

Reply:
I think you are looking at this config as "who provides DNS for Ingress?" Whereas, I have designed this API to indicate "who provides DNS for the cluster?". I think the 2nd question is the most generic form of the question.

We need DNS for Ingress to be functional during cluster bringup so we have a CoreDNS pod that provides that during cluster bring up. When the cluster is up, we do want the user to update their DNS solution for the *.apss entry too. When all steps are completed, the customer's DNS solution resolves API, API-Int and Ingress.

The enhancement proposal https://github.com/openshift/enhancements/pull/1400 talks about this in greater detail.